### ISSUE TITLE
Catch MissingSchema errors that could be raised by requests

### ISSUE THREAD

**Issue by non-Jedi** [matrix-org/matrix-python-sdk#221]
See here for example: https://github.com/matrix-org/matrix-python-sdk/pull/179/files#diff-730c26e17b5455de7bbc524b7a20dc2fR33

requests is an implementation detail, so we shouldn't allow end users to see any of its errors.

**Comment by slipeer** [comment]
Perhaps the best solution would be to check URL in the client initiation code?
Client initialization - is single place to implement this check (against a lot of places where you have to check error from requests).
For example:
```python
from urllib3.util import parse_url
from urllib3.exceptions import LocationParseError
try:
    scheme, auth, host, port, path, query, fragment = parse_url(url)
except LocationParseError as e:
    raise MatrixError("Invalid homeserver url %s" % url)

if not scheme:
    raise MatrixError("No scheme in homeserver url")
```

**Comment by non-Jedi** [comment]
Ya, that could definitely work. Would need to move `MatrixHttpApi.base_url` to something like `MatrixHttpApi._base_url` to make sure people don't modify it at runtime, though. Reality is we have exactly one place where we call into `requests` with variable arguments (in `MatrixHttpApi._send`), so it's probably a lot easier to just catch this error there. It's not like this sdk will ever get used for very long with a `MatrixHttpApi` instantiated but no other methods called, so moving the validation to instantiation doesn't have a huge positive effect. I wouldn't be opposed to either solution though.

**Comment by non-Jedi** [comment]
Closed in #256
